# T-100 Data Acquisition

This notebook documents and verifies the acquisition and initial loading of BTS T-100 International Segment data for the Airline Passenger Demand Forecasting & Capacity Planning project.

In [1]:
import sys

print(sys.executable)
print(sys.version)

c:\Users\PC\airline-demand-capacity-planning\.venv\Scripts\python.exe
3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]


In [2]:
import pandas as pd

print("pandas:", pd.__version__)

pandas: 3.0.5


In [3]:
from pathlib import Path
from zipfile import ZipFile

raw_dir = Path("../data/raw")

matching_files = list(
    raw_dir.glob("T_T100_SEGMENT_ALL_CARRIER_Jan*.zip")
)

if len(matching_files) != 1:
    raise ValueError(
        f"Expected exactly one T-100 Segment All Carriers ZIP, "
        f"found {len(matching_files)}"
    )

raw_path = matching_files[0]

In [4]:
df = pd.read_csv(raw_path)
print('type: ' , type(df))
print('shape: ' , df.shape)

type:  <class 'pandas.DataFrame'>
shape:  (45204, 50)


In [5]:
for i,column in enumerate(df.columns, start=1):
    print(i, column)

1 DEPARTURES_SCHEDULED
2 DEPARTURES_PERFORMED
3 PAYLOAD
4 SEATS
5 PASSENGERS
6 FREIGHT
7 MAIL
8 DISTANCE
9 RAMP_TO_RAMP
10 AIR_TIME
11 UNIQUE_CARRIER
12 AIRLINE_ID
13 UNIQUE_CARRIER_NAME
14 UNIQUE_CARRIER_ENTITY
15 REGION
16 CARRIER
17 CARRIER_NAME
18 CARRIER_GROUP
19 CARRIER_GROUP_NEW
20 ORIGIN_AIRPORT_ID
21 ORIGIN_AIRPORT_SEQ_ID
22 ORIGIN_CITY_MARKET_ID
23 ORIGIN
24 ORIGIN_CITY_NAME
25 ORIGIN_STATE_ABR
26 ORIGIN_STATE_FIPS
27 ORIGIN_STATE_NM
28 ORIGIN_COUNTRY
29 ORIGIN_COUNTRY_NAME
30 ORIGIN_WAC
31 DEST_AIRPORT_ID
32 DEST_AIRPORT_SEQ_ID
33 DEST_CITY_MARKET_ID
34 DEST
35 DEST_CITY_NAME
36 DEST_STATE_ABR
37 DEST_STATE_FIPS
38 DEST_STATE_NM
39 DEST_COUNTRY
40 DEST_COUNTRY_NAME
41 DEST_WAC
42 AIRCRAFT_GROUP
43 AIRCRAFT_TYPE
44 AIRCRAFT_CONFIG
45 YEAR
46 QUARTER
47 MONTH
48 DISTANCE_GROUP
49 CLASS
50 DATA_SOURCE


In [6]:
key_columns = ['UNIQUE_CARRIER', 'YEAR', 'MONTH', 'ORIGIN', 'DEST', 'PASSENGERS', 'SEATS', 'DEPARTURES_SCHEDULED', 'DEPARTURES_PERFORMED', 'DISTANCE']
print(df[key_columns].dtypes)

UNIQUE_CARRIER              str
YEAR                      int64
MONTH                     int64
ORIGIN                      str
DEST                        str
PASSENGERS              float64
SEATS                   float64
DEPARTURES_SCHEDULED    float64
DEPARTURES_PERFORMED    float64
DISTANCE                float64
dtype: object


In [7]:
df[key_columns].head()

,UNIQUE_CARRIER,YEAR,MONTH,ORIGIN,DEST,PASSENGERS,SEATS,DEPARTURES_SCHEDULED,DEPARTURES_PERFORMED,DISTANCE
0,NEW,2024,1,BID,WST,0.0,0.0,0.0,0.0,17.0
1,NEW,2024,1,WST,BID,0.0,0.0,0.0,0.0,17.0
2,9K,2024,1,MVY,HYA,0.0,0.0,0.0,0.0,26.0
3,DL,2024,1,JFK,PHL,0.0,0.0,0.0,0.0,94.0
4,9K,2024,1,ACK,JFK,0.0,0.0,0.0,0.0,199.0


In [8]:
print('YEAR: ', df['YEAR'].unique())
print('MONTH: ', df['MONTH'].unique())

YEAR:  [2024]
MONTH:  [1]


In [9]:
print(set(key_columns) - set(df.columns.tolist()))

set()


## Initial Acquisition Verification

- Source: U.S. DOT Bureau of Transportation Statistics (BTS)
- Table: T-100 Segment (All Carriers)
- Carrier scope: ALL CARRIERS
- Requested period: January 2024
- Raw archive: T_T100I_SEGMENT_ALL_CARRIER_20260907_133912.zip
- File inside archive: T_T100_SEGMENT_ALL_CARRIER.csv
- DataFrame shape: (45204, 50)
- Number of columns: 50
- Required project fields present: Yes

## Source Field Mapping
|BTS CSV Column| BTS Meaning | Project Concept|
|--------------|--------------|----------------|
|UNIQUE_CARRIER| Unique Carrier Code | Carrier Code|
|YEAR| Year |Year of observation|
|MONTH| Month Of Flight|Month of observation|
|ORIGIN| Origin Airport|Directional route origin|
|DEST| Destination Airport| Directional route destination|
|PASSENGERS| non-stop segment passengers transported | Observed passenger volume|
|SEATS| Available Seats| Historical available seat capacity
| DEPARTURES_SCHEDULED | Departures Scheduled | Scheduled flight departures |
| DEPARTURES_PERFORMED | Departures Performed | Actual performed departures |
| DISTANCE | Distance between airports (miles) | Segment distance |

## Acquisition Method

The verification sample was obtained from the U.S. DOT Bureau of Transportation Statistics (BTS) TranStats download interface using the T-100 Segment (All Carriers) table.

Source reference:

https://www.transtats.bts.gov/DL_SelectFields.aspx?QO_fu146_anzr=Nv4+Pn44vr45&gnoyr_VQ=FMG

Download selections:

- Geography: unrestricted
- Year: 2024
- Period: January
- Fields: all available fields
- Carrier scope: all carriers
- Download format received: ZIP archive containing one CSV file
- Acquisition date: 2026-09-07

The original BTS archive is preserved unchanged in:

`data/raw/T_T100_SEGMENT_ALL_CARRIER_20260907_133912.zip`

The archive contains:

`T_T100_SEGMENT_ALL_CARRIER.csv`

Raw files under `data/raw/` are excluded from Git tracking.

The CSV is loaded directly from the ZIP archive with Pandas, so the raw BTS archive does not need to be extracted or modified.